peekingduck/training/src/model/tensorflow_model.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 AI Singapore
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+import tensorflow as tf
+from omegaconf import DictConfig
+
+from src.model.tensorflow_base import TFModelFactory
+from src.utils.tf_model_utils import set_trainable_layers
+
+logger = logging.getLogger("TF Model")  # pylint: disable=invalid-name
+logging.basicConfig(level=logging.INFO)
+
+
+class TFClassificationModelFactory(TFModelFactory):
+    """Generic TensorFlow image classification model."""
+
+    @classmethod
+    def create_model(cls, model_cfg: DictConfig) -> tf.keras.Model:
```

Comment:
Same issue as the `create_head`, consider to have the same set of arguments in the base class as well.